lib/apiservers/portlayer/restapi/handlers/interaction_handlers.go

```diff
@@ -361,38 +361,55 @@ func (d *FlushingReader) AddFlusher(flusher GenericFlusher) {
 // ignored.  This reader does not care about the init sequeunce.  The init sequence
 // maybe used by the higher level interaction, which in this case is the Swagger
 // establishing initial connection for stdin.
+//
+// Panics if the buf is smaller than the initBytes
 func (d *FlushingReader) readDetectInit(buf []byte) (int, error) {
 	// fast path - len(nil) return 0
 	if len(d.initBytes) == 0 {
 		return d.Read(buf)
 	}
 
-	var er error
+	// make sure we have enough room
+	if len(buf) < len(d.initBytes) {
+		panic("Read buffer is smaller than the initialization byte sequence")
+	}
+
+	var err error
+
 	initLen := len(d.initBytes)
 	total := 0
+	upto := 0
 	for total < initLen {
-		nr, er := d.Read(buf)
+		nr, err := d.Read(buf[total:])
 		total += nr
 
-		if bytes.Compare(d.initBytes[0:total], buf[0:total]) != 0 {
+		log.Debugf("Primer bytes read %d byte, err is %q buf is %q", nr, err, buf[0:total])
```

Comment:
Is it really necessary to log whole buffer? it can be HUGE. I would consider limiting a slice size for logging purposes if it exceed 200 or something bytes.